src/main/java/edu/harvard/iq/dataverse/engine/command/impl/CreateDataverseCommand.java

```diff
@@ -73,25 +82,48 @@ public Dataverse execute(CommandContext ctxt) throws CommandException {
         if (created.getDataverseType() == null) {
             created.setDataverseType(Dataverse.DataverseType.UNCATEGORIZED);
         }
-        
+
         if (created.getDefaultContributorRole() == null) {
             created.setDefaultContributorRole(ctxt.roles().findBuiltinRoleByAlias(DataverseRole.EDITOR));
         }
-        
+
         // @todo for now we are saying all dataverses are permission root
         created.setPermissionRoot(true);
         
         if ( ctxt.dataverses().findByAlias( created.getAlias()) != null ) {
             throw new IllegalCommandException("A dataverse with alias " + created.getAlias() + " already exists", this );
         }
-        
+
         // Save the dataverse
         Dataverse managedDv = ctxt.dataverses().save(created);
 
         // Find the built in admin role (currently by alias)
         DataverseRole adminRole = ctxt.roles().findBuiltinRoleByAlias(DataverseRole.ADMIN);
         String privateUrlToken = null;
         ctxt.roles().save(new RoleAssignment(adminRole, getRequest().getUser(), managedDv, privateUrlToken));
+        // Add additional admins if inheritance is set
+        if (ctxt.settings().isTrueForKey(SettingsServiceBean.Key.InheritParentAdmins, false)) {
+            List<RoleAssignment> assignedRoles = ctxt.roles().directRoleAssignments(owner);
+            for (RoleAssignment role : assignedRoles) {
```

Comment:
Again here.

Reply:
@oscardssmith - This gets the full list of roles on the parent Dataverse and then filters to find just the ones that are admin roles. I did not see a DataverseRoleServiceBean call to just get direct roles of one type. Is there one? Or are you suggesting one is needed?